arrow-fx-coroutines/src/main/kotlin/arrow/fx/coroutines/stream/Compiler.kt

```diff
@@ -106,7 +106,7 @@ internal fun <O> interruptBoundary(
       when (val close = view.step) {
         is Pull.Eval.CloseScope -> Pull.Eval.CloseScope(
           close.scopeId,
-          Pair(interruptedScope, close.interruptedScope?.second),
+          Pair(interruptedScope, interruptedError),
```

Comment:
We were passing along the wrong exception here.
This fixes https://github.com/arrow-kt/arrow-fx/pull/210/files#diff-ce71ba0355e5832a46e62f6d1ee87360R249.

The problem there was that we need to pass along the `interruptedError`, to re-raise the error later on.


If we look at how `flatMap` is implemented, we can see that it does `unconsOrNull`, which pulls the next `Chunk<O>.

Then it'll transform the `tail` with this method `interruptBoundary`:
 - in the case the `tail` was not finished (`Pull.Result`), it will either `Intterupt` or `Fail`
 - When we encounter `CloseScope`, we're going to set our current  `Interuption Context` to our `CloseScope`.
 - When we encounter a case that produces a new value i.e. `Effect` we just evaluate it and move to the next step.

Reply:
Right, I think I should have been more clear with my question. Isn't there a chance for interruptedError to be `null` here for the `CloseScope`? what'd be the case when that happens?

Reply:
Ah, yes there is a chance for it to be `null`.
If it's `null` then we consider it a graceful interruption.

`Stream.sleep(1.second).interruptWhen { Right(sleep(500.milliseconds)) }`
 vs

`Stream.sleep(1.second).interruptWhen { sleep(500.milliseconds); Left(RuntimeException("Boom!")) }`

Reply:
Perfect, thanks for clarifying!